build.gradle

```diff
@@ -186,6 +186,11 @@ allprojects {
     else
       options.links "https://docs.oracle.com/javase/8/docs/api/"
   }
+
+  clean {
```

Comment:
Should we add `cleanTest` for consistency? for example:
```
  cleanTest {
    delete "${projectDir}/src/generated-test"
  }
```

Reply:
I wasn't aware of cleanTest, so I had to look at the documentation: https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_tasks

It seems that cleanTest means to delete the test results, which isn't quite the right situation to use this. That would delete the generated sources, the test results, but _not_ the test jar.

If i wanted to follow the Java plugin's conventions, i would probably add a cleanProcessMessages and cleanProcessTestMessages and have `clean` depend on them. Do you prefer that over applying them to `clean` directly?

Another alternative would be to move the generated sources into the `build` directory so they get automatically cleaned up, but that's a more invasive change.